File: app/src/main/java/it/amonshore/comikkua/ui/releases/ReleaseItemDetails.java
package it.amonshore.comikkua.ui.releases;

import androidx.annotation.Nullable;
import androidx.recyclerview.selection.ItemDetailsLookup;

public class ReleaseItemDetails extends ItemDetailsLookup.ItemDetails<Long> {

    private int mPosition;
    private Long mSelectionKey;

    public ReleaseItemDetails(int position, Long selectionKey) {
        mPosition = position;
        mSelectionKey = selectionKey;
    }

    @Override
    public int getPosition() {
        return mPosition;
    }

    @Nullable
    @Override
    public Long getSelectionKey() {
        return mSelectionKey;
    }
}

File: app/src/main/java/it/amonshore/comikkua/ui/releases/ReleaseItemDetails.kt
package it.amonshore.comikkua.ui.releases

import androidx.recyclerview.selection.ItemDetailsLookup.ItemDetails

data class ReleaseItemDetails(
    private val position: Int,
    private val selectionKey: Long
) : ItemDetails<Long>() {

    override fun getPosition(): Int {
        return position
    }

    override fun getSelectionKey(): Long {
        return selectionKey
    }
}